### PR TITLE
Make AnimatableSlider inherit from UISlider and add designable images

### PR DESCRIPTION
**Pull request by phimage**
`AnimatableSlider` do not extends `UISlider`, only `UIView`

thumb, minimum and maximum track images added has designable objects in storyboard

PS: 
I try some experiment on slider size by overriding `trackRect()`

```
 @IBInspectable var trackHeight: CGFloat = CGFloat.nan
    
    override func trackRect(forBounds bounds: CGRect) -> CGRect {
       // add here if not trackHeight.isNan
        return CGRect(origin: bounds.origin, size: CGSize(width:bounds.width, height:trackHeight))
    }
 // or
    override func trackRect(forBounds bounds: CGRect) -> CGRect {
       let rect = super.trackRect(forBounds: bounds)
       return ...
    }
```
There is some bug when using maximum and minimum images
to fix some other bugs in storyboard  I use in `trackRect()`
```

#if TARGET_INTERFACE_BUILDER
 // do some stuff to adjust storyboard
#endif
```

### CODE REVIEW

File: IBAnimatable/SliderImagesDesignable.swift [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+  var maximumTrackImage: UIImage? { get set }
+
+  /**
+   * The maximum track image when slider state is `highlighted`
+   */
+  var maximumTrackHighlightedImage: UIImage? { get set }
+
+}
+
+extension SliderImagesDesignable where Self: UISlider {
```
Comment (JakeLin) [inline IBAnimatable/SliderImagesDesignable.swift]: I think we need to make it `public` for the others can use those extension methods outside of `IBAnimatable` framework.

File: IBAnimatable/SliderImagesDesignable.swift [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+  func configureThumbImage() {
+    self.setThumbImage(thumbImage, for: .normal)
```
Comment (JakeLin) [inline IBAnimatable/SliderImagesDesignable.swift]: Shall we remove all `self`?

**Review by JakeLin — APPROVED** [review]
I like you PR very much with a lot of documentation comments 👍👍👍. I think this PR is ready to go. It will be greate if you can modify the CHANGELOG.md file. It is very simple like https://github.com/IBAnimatable/IBAnimatable/blob/master/CHANGELOG.md#enhancements

One thing I need to confirm with @tbaranes on why did we inherit from `UIView`. If he is OK, we will merge it. Thanks again 💪

**Comment by JakeLin** [comment]
@phimage thank you very much for your PR. Please have a look at my small comments.

@tbaranes In #228, we created `AnimatableSlider`. I know we want to have fully customizable  `AnimatableSlider` and `UISlider` can't give us that 😕. Therefore, we let it inherit from `UIView` instead of `UISlider`. @phimage found out that and made `AnimatableSlider` to inherit from `UISlider`. I think that is better for now since the current `AnimatableSlider` is more like `AnimatableView`. 


If we want to have fully customizable  `AnimatableSlider` in the future, we may create another class like `AnimatableCustomSlider` to inherit from `UIView`. Can I have your opinion on this PR?  thanks.

File: IBAnimatable/SliderImagesDesignable.swift (hunk 2 above)
Comment (tbaranes) [reply IBAnimatable/SliderImagesDesignable.swift]: +1, also all the others below 😬

**Review by tbaranes — APPROVED** [review]

**Comment by tbaranes** [comment]
I agree with you @JakeLin, that PR is a good move since it's making `AnimatableSlider` more _slider_. I don't really remember why we choose `UIView` instead of `UISlider` in #228, but I can't see any reasons. Truth is when we will do the custom sliders, it won't really be related to `AnimatableSlider`, so... who cares?

@phimage Good job on this!

**Comment by phimage** [comment]
Ok for the changes (`public` and `self`)
For changelog I make two lines? one in "api breaking changes" or not?

I will fix also some changelog issue with markdown comma

**Comment by JakeLin** [comment]
@phimage well done. It is in 🎉, thanks a lot. It will be in version 4. @tbaranes thanks for your feedback.

**Pull request merged**